src/components/solvers/ProgressHandler.tsx

```diff
@@ -1,33 +1,70 @@
-import { Box, Center, HStack, VStack } from "@chakra-ui/react";
-import React, { useState } from "react";
+import { Box, Center, HStack } from "@chakra-ui/react";
+import React, { useEffect, useState } from "react";
+import { History } from "./History";
 import { GoButton } from "./buttons/GoButton";
-import { postProblem } from "../../api/ToolboxAPI";
+import { fetchSolution, postProblem } from "../../api/ToolboxAPI";
 import { SolutionView } from "./SolutionView";
 import { Container } from "../Container";
 import { Solution } from "../../api/data-model/Solution";
 import { SolveRequest } from "../../api/data-model/SolveRequest";
 import { SolverPicker } from "./SolverPicker";
 
+export interface ProblemState<T> {
+  /**
+   * Problem input
+   */
+  content: T;
+  /**
+   * Ids of the solutions of the problem input per problem type
+   */
+  solutionIds: ProblemTypeSolutionId;
```

Comment:
There is a serialization issue here but not the one we had when passing a map to the server.
When storing something in the local storage, we just store a string. When deserializing the string, the type information of the map won't persist because of TypeScript limitations which causes that you can't call any functions on the object, as only the primitive object data is loaded but not the type information which are lost.